File: modules/foreground-service/src/ForegroundServiceModule.ts
import { NativeModule, requireNativeModule } from 'expo';
import type { ForegroundServiceNativeModule } from './ForegroundService.types';

declare class ForegroundServiceModule extends NativeModule implements ForegroundServiceNativeModule {
  startService(title?: string, message?: string): Promise<boolean>;
  stopService(): Promise<boolean>;
  updateNotification(title: string, message: string): Promise<boolean>;
  acquireWakeLock(timeoutMs?: number): Promise<boolean>;
  releaseWakeLock(): Promise<boolean>;
  isServiceRunning(): Promise<boolean>;
  isWakeLockActive(): Promise<boolean>;
  addListener(eventName: string, listener: (event: any) => void): { remove: () => void };
  removeListeners(count: number): void;
}

export default requireNativeModule<ForegroundServiceModule>('ForegroundService');

File: modules/foreground-service/ios/ForegroundServiceModule.swift
import ExpoModulesCore

public class ForegroundServiceModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ForegroundService")

    Events("onMarkSafe")

    AsyncFunction("startService") { (title: String?, message: String?) -> Bool in
      return false
    }

    AsyncFunction("stopService") { () -> Bool in
      return false
    }

    AsyncFunction("updateNotification") { (title: String, message: String) -> Bool in
      return false
    }

    AsyncFunction("acquireWakeLock") { (timeoutMs: Double?) -> Bool in
      return false
    }

    AsyncFunction("releaseWakeLock") { () -> Bool in
      return false
    }

    AsyncFunction("isServiceRunning") { () -> Bool in
      return false
    }

    AsyncFunction("isWakeLockActive") { () -> Bool in
      return false
    }
  }
}
